src/components/Experiment.vue

```diff
@@ -141,14 +121,23 @@ export default {
     }
 
     return {
+      // config
+      ...this.$options.magpie,
+      id: this.$options.magpie.experimentId,
+
       currentScreen: 0,
       results: {},
       currentTrial: {},
       trial,
       mousetrackingTime: [0],
       mousetrackingX: [0],
       mousetrackingY: [0],
-      mousetrackingStartTime: 0
+      mousetrackingStartTime: 0,
+      socket: new Socket(
```

Comment:
Another thing to consider is that we may want to make sure that the component doesn't throw an error when the experiment is not interactive... i.e. the experimenter will then not provide a socketURL in the config

Reply:
Of course, I guess if you don't call `socket.connect()`, nothing will actually happen, since here we only have a Socket object which holds the configuration, but which doesn't actually do any real operation.

Reply:
Good point, I'll make this optional.